debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerDmlParser.java

```diff
@@ -0,0 +1,473 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.oracle.logminer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.debezium.DebeziumException;
+import io.debezium.connector.oracle.logminer.valueholder.LogMinerColumnValue;
+import io.debezium.connector.oracle.logminer.valueholder.LogMinerColumnValueImpl;
+import io.debezium.connector.oracle.logminer.valueholder.LogMinerDmlEntry;
+import io.debezium.connector.oracle.logminer.valueholder.LogMinerDmlEntryImpl;
+import io.debezium.data.Envelope;
+
+/**
+ * @author Chris Cranford
+ */
+public class LogMinerDmlParser {
+
+    private static final String SINGLE_QUOTE = "'";
+    private static final String NULL = "NULL";
+    private static final String INSERT_INTO = "insert into ";
+    private static final String UPDATE = "update ";
+    private static final String DELETE = "delete ";
+    private static final String AND = "and ";
+    private static final String OR = "or ";
+    private static final String SET = " set ";
+    private static final String WHERE = " where ";
+    private static final String VALUES = " values ";
+
+    /**
+     * Parse a DML SQL statement.
+     *
+     * @param sql the sql statement
+     * @return the parsed DML entry record or {@code null} if the SQL was not parsed
+     */
+    public LogMinerDmlEntry parse(String sql) {
+        if (sql != null && sql.length() > 0) {
+            switch (sql.charAt(0)) {
+                case 'i':
+                    return parseInsert(sql);
+                case 'u':
+                    return parseUpdate(sql);
+                case 'd':
+                    return parseDelete(sql);
+            }
+        }
+        return null;
```

Comment:
Should we throw an exception rather? What did the JsqlParser impl do in this case?

Reply:
So the other parser implementation was designed to catch any parse exception, log an error, and then return `null` which caused the processing loop to skip the event and move onto the next event as if nothing happened.  Would you agree we probably should throw some type of `DmlParserException` which would cause the connector task to fail?